packages/headless/docs.config.json

```diff
@@ -0,0 +1,279 @@
+        },
+        "others": [
+          {
+            "name": "FacetProps",
+            "source": "src/controllers/facets/facet/headless-facet.ts"
+          },
+          {
+            "name": "FacetOptions",
+            "source": "src/controllers/facets/facet/headless-facet-options.ts"
+          }
+        ]
+      },
+      "state": "FacetState",
+      "related": [],
+      "subcontrollers": [
+        {
+          "name": "FacetSearch",
+          "initialize": {
+            "initializer": {
+              "name": "buildFacetSearch",
+              "source": "src/controllers/facets/facet-search/specific/headless-facet-search.ts"
+            },
+            "others": [
+              {
+                "name": "FacetSearchProps",
+                "source": "src/controllers/facets/facet-search/specific/headless-facet-search.ts"
+              },
+              {
+                "name": "FacetSearchOptions",
+                "source": "src/features/facets/facet-search-set/facet-search-request-options.ts"
+              }
+            ]
+          },
+          "state": "none",
+          "related": []
+        }
+      ]
+    },
+    {
+      "name": "ResultList",
+      "initialize": {
+        "initializer": {
+          "name": "buildResultList",
+          "source": "src/controllers/result-list/headless-result-list.ts"
+        },
+        "others": []
+      },
+      "state": "ResultListState",
+      "related": [
+        {
+          "name": "Result",
+          "source": "src/api/search/search/result.ts"
+        },
+        {
+          "name": "ResultTemplateCondition",
+          "source": "src/features/result-templates/result-templates.ts"
+        },
+        {
+          "name": "ResultTemplate",
+          "source": "src/features/result-templates/result-templates.ts"
+        },
+        {
+          "name": "ResultTemplatesManager",
+          "source": "src/features/result-templates/result-templates-manager.ts"
+        },
+        {
+          "name": "buildResultTemplatesManager",
+          "source": "src/features/result-templates/result-templates-manager.ts"
+        },
+        {
+          "name": "getResultProperty",
+          "source": "src/features/result-templates/result-templates-helpers.ts"
+        },
+        {
+          "name": "fieldsMustBeDefined",
+          "source": "src/features/result-templates/result-templates-helpers.ts"
+        },
+        {
+          "name": "fieldsMustNotBeDefined",
+          "source": "src/features/result-templates/result-templates-helpers.ts"
+        },
+        {
+          "name": "fieldMustMatch",
+          "source": "src/features/result-templates/result-templates-helpers.ts"
+        }
+      ]
+    },
+    {
+      "name": "SearchBox",
+      "initialize": {
+        "initializer": {
+          "name": "buildSearchBox",
+          "source": "src/controllers/search-box/headless-search-box.ts"
+        },
+        "others": [
```

Comment:
_Others_ are within the _initialize_ section, whereas _related_ is not. They need to be explicitly in the configuration because we need their source (options often aren't in the same file as the build function).

Reply:
Hmm, I see, I would suggest `args` for the property name, but the plural would be inaccurate, since there is only a single props object. Two questions:

1. How is the engine (the first arg) resolved, since it is in a different file too?
2. Is the order of the entries in the array important? How do we resolve that `SearchBoxOptions` falls under `SearchBoxProps`, and not vice-versa?

Reply:
Args for any function are listed as part of its call signature. The call signature contains the name of the arg and the type of the arg, but it doesn't actually resolve the type, it just lists it. You have to go to that arg's own entry, wherever that may be, in order to resolve the whole type. So the engine arg is "resolved" here in that we know its name and that it is type Engine with type arguments, but we don't know anything about the actual Engine type itself. That's part of the engine extraction which hasn't been done yet; at that point, the Engine will also have to be added to the config in the engine section.

As for the order of the entries, I envision it determining the order that they are displayed on a docs page, yes; however, it doesn't effect knowing which one falls under the other. When we list out `SearchBoxProps`, it has one property, `options`, whose type is `SearchBoxOptions`. That's how users will know that the options fall under the props.